### ISSUE TITLE
Regex matcher misbehaving on modern linux

### ISSUE THREAD

**Issue by denandz** [denandz/fuzzotron#18]
Regex matcher functionality seems to be misbehaving on Fedora 40 on ARM64. 

```
./fuzzotron --radamsa --directory testcases  -h 127.0.0.1 -p 80 -P udp -m ./foo.log -r 'segfault' -o output
```

does not trip after running `echo 'weh segfault wah' >> foo.log`

**Comment by denandz** [comment]
Tested on an x86 box too - `fopen()/fseek()/getline()` in `monitor.c` doesn't end up returning `>0` from the `getline()` call - log file monitoring needs revisiting and an overhaul to run on 6.x kernels properly.

**Comment by denandz** [comment]
After some messing around - looks like this is due to some changes in glibc. Moving to `open()/read()` would be one option, or replicating https://github.com/coreutils/coreutils/blob/master/src/tail.c logic

**Comment by denandz** [comment]
fixed by commit b2cb5c0
